### DISCUSSION TITLE
Car Status?

### DISCUSSION THREAD

**Discussion by grahamck** [synchronizing/bluelink#5] [Ideas]
Thanks for this - it works really well. 

I wonder if there is a way to grab that car status that bluelink provides. If doors are open, if it is running, etc. 

That would be very useful (i think). :-)

**Comment by synchronizing** [comment]
Hey @grahamck. It's likely possible, but would require some digging on the API front. I unfortunately no longer own my old Hyundai, so I can't test it out.

**Reply by grahamck** [reply]
OK - thanks for the reply - I can do this with another package so it's not a big deal. Just like python more than java. ;-)

**Reply by synchronizing** [reply]
Mind sharing it? I can take a look at what APIs they are hitting and see if I can bring this over to this.

**Reply by grahamck** [reply]
I am getting status from the "bluelinky" package published here:

https://github.com/Hacksore/bluelinky

I am not in any way a java expert - but the "vehicle.status"  call returns the status of the vehicle in a couple of different formats. 

The refresh parameter set to true  appears to go to the car and poll. If false it appears to pull the latest data from hyundai's server. I have experimented with this a bit and if it is set to false it does not appear to count towards the limit put on bluelink calls in the US (which locks me out after 50 consecutive calls with it set to "true").  Also, if the car is locked or unlocked it appears to push that data to the server. That is as far as I have gone so far in testing. 


 const response = await vehicle.status({ refresh: true, parsed: false });
    console.log(response);


I send the output to a file and wrote some perl scripts to parse it. 

Hope this helps. :-) Thanks!

**Reply by synchronizing** [reply]
Thank you for sharing it. I looked through the codebase, and they are using a completely different API endpoint, unfortunately. For us to get this working we would have to either port the entire project over to Python, or have someone explore the API this project uses and see if they can find this info.

**Reply by synchronizing** [reply]
I went through some of projects old commits and noticed that they _were_ using the API this project is using. I'll have a friend test out the endpoint they were reaching out to later today and see if we get anything.
